### PR TITLE
Add travis configuration for Pulp 3

### PR DESCRIPTION
**Pull request by jeremycline**
This includes some re-arrangement of code as well as the necessary updates to the .travis.yml file. This is because I don't want install pulp.server and its dependencies.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,16 @@
 language: python
 python:
     - "2.7"
+    - "3.4"
+    - "3.5"
+addons:
+    postgresql: "9.3"
```
Comment (jeremycline) [inline .travis.yml]: I couldn't remember what version we were targeting, but I assume we should work with any recent version.

File: platform/pulp/platform/logs.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
+"""
```
Comment (jeremycline) [inline platform/pulp/platform/logs.py]: I left the other version of this file in server/. I'm interested to see whether others think we should try to keep the Pulp 2 setup working with a Pulp 3 checkout or not. If not (which is what I'm in favor of), I'll just move the server/ version here.

File: platform/pulp/platform/models/progress.py [hunk 3]
```diff
@@ -19,6 +19,19 @@
 _logger = logging.getLogger(__name__)
 
 
+def get_current_task_id():
```
Comment (jeremycline) [inline platform/pulp/platform/models/progress.py]: I copied this this function out of pulp.server.async.tasks, but I don't know that it should live here. @bmbouter?

**Comment by jeremycline** [comment]
A few things to note here: someone needs to flip the repository setting switches to start running this, and we are already incompatible with Python 3 in platform/ so tests will fail until we fix that.

File: platform/pulp/platform/models/progress.py (hunk 3 above)
Comment (bmbouter) [inline platform/pulp/platform/models/progress.py]: I'm actually moving this function as part of the conversion of the tasking code to use the new models. What if we let my PR clean up this import and you continue to use it from `pulp.server.async.tasks` ?

File: .travis.yml [hunk 4]
```diff
@@ -1,7 +1,16 @@
 language: python
 python:
     - "2.7"
+    - "3.4"
+    - "3.5"
```
Comment (bmbouter) [inline .travis.yml]: :+1:

**Comment by bmbouter** [comment]
I posted just one piece of feedback, but I'm adding the LGTM now because I don't need to re-review.

File: platform/pulp/platform/models/progress.py (hunk 3 above)
Comment (jeremycline) [inline platform/pulp/platform/models/progress.py]: Cool. I'll change it back, but I won't install the ``pulp.server`` package. The tests will fail until your PR is merged, but the Python 3.y tests fail in any case (I've got a branch ready for that as well).

**Pull request merged**